### PR TITLE
Use more progressive disclosure for Taxonomy tagging

### PR DESCRIPTION
**Pull request by cbaines**
Allow each level of the Topic Taxonomy to be expanded.

This change also allows the level one taxons to be selected and
unselected, which wasn't possible in the previous design.

taxonomy-progressive-disclosure

### CODE REVIEW

**Comment by 1pretz1** [comment]
Bug has been fixed! 🐛

File: app/assets/javascripts/admin/modules/taxonomy_tree_checkboxes.js [hunk 1]
```diff
@@ -2,44 +2,49 @@
   "use strict";
 
   Modules.TaxonomyTreeCheckboxes = function() {
-    var ancestors = function(element) {
-      var $parents = $(element).parents('.topics,.topic-tree');
-      return $parents.prev('p').find('input[type="checkbox"]');
+    var ancestors = function($element) {
+      var parentContentID = $element.data("parent-content-id");
+      console.log("parentContentID " + parentContentID);
```
Comment (koetsier) [inline app/assets/javascripts/admin/modules/taxonomy_tree_checkboxes.js]: do we need this line?

File: app/assets/javascripts/admin/modules/taxonomy_tree_checkboxes.js [hunk 2]
```diff
@@ -2,44 +2,49 @@
...
+      return $("#topic-tree-" + $element.attr("id")).find('input[type="checkbox"]');
     };
 
     /**
      Check all ancestor topics.
      */
-    var checkAncestors = function(element) {
-      ancestors(element).prop('checked', true);
+    var checkAncestors = function($element) {
+      console.log("ancestors", ancestors($element));
```
Comment (koetsier) [inline app/assets/javascripts/admin/modules/taxonomy_tree_checkboxes.js]: do we need this?

File: app/assets/javascripts/admin/modules/taxonomy_tree_checkboxes.js (hunk 1 above)
Comment (1pretz1) [reply app/assets/javascripts/admin/modules/taxonomy_tree_checkboxes.js]: nope

File: app/assets/javascripts/admin/modules/taxonomy_tree_checkboxes.js (hunk 2 above)
Comment (1pretz1) [reply app/assets/javascripts/admin/modules/taxonomy_tree_checkboxes.js]: and nope again

**Comment by 1pretz1** [comment]
The two console outputs have been removed!

**Pull request merged**